src/main/java/core/basesyntax/service/UserValidator.java

```diff
@@ -0,0 +1,72 @@
+package core.basesyntax.service;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exeptions.RegistrationException;
+import core.basesyntax.model.User;
+
+public class UserValidator {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final int MIN_LOGIN_LENGTH = 6;
+
+    private final StorageDaoImpl storageDao;
+
+    public UserValidator() {
+        storageDao = new StorageDaoImpl();
+    }
+
+    public void validateAge(User user) throws RegistrationException {
+        if (user.getAge() == null) {
+            throw new RegistrationException("Age is null. "
+                    + "Enter correct age.");
+        }
+        if (isAgeHighEnough(user)) {
+            throw new RegistrationException("Incorrect age. "
+                    + "Enter age between 18 and 150.");
+        }
+    }
+
+    public void validateLogin(User user) throws RegistrationException {
+        if (user.getLogin() == null) {
+            throw new RegistrationException("Login is null. "
+                    + "Enter correct login.");
+        }
+        if (isLoginLongEnough(user)) {
+            throw new RegistrationException("Login is too short."
+                    + "Enter login longer than 6 characters.");
+        }
+        if (isUserLoginAlreadyExist(user)) {
+            throw new RegistrationException("There is user with the same login"
+                    + "Enter login that not exist.");
+        }
+    }
```

Comment:
inside validationLogin method you are using only `String login`, why do you need to pass User every time to method?
the same for other validation methods
```suggestion
    public void validateLogin(String login) throws RegistrationException {
        if (user.getLogin() == null) {
            throw new RegistrationException("Login is null. "
                    + "Enter correct login.");
        }
        if (isLoginLongEnough(user)) {
            throw new RegistrationException("Login is too short."
                    + "Enter login longer than 6 characters.");
        }
        if (isUserLoginAlreadyExist(user)) {
            throw new RegistrationException("There is user with the same login"
                    + "Enter login that not exist.");
        }
    }
```